sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientCertificateCredential.java

```diff
@@ -35,16 +36,19 @@
      * @param tenantId the tenant ID of the application
      * @param clientId the client ID of the application
      * @param certificatePath the PEM file or PFX file containing the certificate
+     * @param certificate the PEM or PFX certificate
      * @param certificatePassword the password protecting the PFX file
      * @param identityClientOptions the options to configure the identity client
      */
-    ClientCertificateCredential(String tenantId, String clientId, String certificatePath, String certificatePassword,
-                                IdentityClientOptions identityClientOptions) {
-        Objects.requireNonNull(certificatePath, "'certificatePath' cannot be null.");
+    ClientCertificateCredential(String tenantId, String clientId, String certificatePath, InputStream certificate,
```

Comment:
Why do we have both the path to the file and the InputStream? If the inputstream is given, then we don't need the path right? Maybe have separate overloads - one with input stream and one with file path.